src/Worker/AzureManaged/DurableTaskSchedulerWorkerOptions.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.ComponentModel.DataAnnotations;
+using Azure.Core;
+using Azure.Identity;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace Microsoft.DurableTask;
+
+/// <summary>
+/// Options for configuring the Durable Task Scheduler.
+/// </summary>
+public class DurableTaskSchedulerWorkerOptions
+{
+    /// <summary>
+    /// Gets or sets the endpoint address of the Durable Task Scheduler resource.
+    /// Expected to be in the format "https://{scheduler-name}.{region}.durabletask.io".
+    /// </summary>
+    [Required(ErrorMessage = "Endpoint address is required")]
+    public string EndpointAddress { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the name of the task hub resource associated with the Durable Task Scheduler resource.
+    /// </summary>
+    [Required(ErrorMessage = "Task hub name is required")]
+    public string TaskHubName { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Gets or sets the credential used to authenticate with the Durable Task Scheduler task hub resource.
+    /// </summary>
+    public TokenCredential? Credential { get; set; }
+
+    /// <summary>
+    /// Gets or sets the resource ID of the Durable Task Scheduler resource.
+    /// The default value is https://durabletask.io.
+    /// </summary>
+    public string ResourceId { get; set; } = "https://durabletask.io";
+
+    /// <summary>
+    /// Gets or sets the worker ID used to identify the worker instance.
+    /// The default value is a string containing the machine name, process ID, and a unique identifier.
+    /// </summary>
+    public string WorkerId { get; set; } = $"{Environment.MachineName},{Environment.ProcessId},{Guid.NewGuid():N}";
+
+    /// <summary>
+    /// Gets or sets a value indicating whether to allow insecure channel credentials.
+    /// This should only be set to true in local development/testing scenarios.
+    /// </summary>
+    public bool AllowInsecureCredentials { get; set; }
+
+    /// <summary>
+    /// Creates a new instance of <see cref="DurableTaskSchedulerWorkerOptions"/> from a connection string.
+    /// </summary>
+    /// <param name="connectionString">The connection string to parse.</param>
+    /// <returns>A new instance of <see cref="DurableTaskSchedulerWorkerOptions"/>.</returns>
+    public static DurableTaskSchedulerWorkerOptions FromConnectionString(string connectionString)
+    {
+        return FromConnectionString(new DurableTaskSchedulerConnectionString(connectionString));
+    }
+
+    /// <summary>
+    /// Creates a gRPC channel for communicating with the Durable Task Scheduler service.
+    /// </summary>
+    /// <returns>A configured <see cref="GrpcChannel"/> instance that can be used to make gRPC calls.</returns>
+    public GrpcChannel CreateChannel()
```

Comment:
If my previous comment about code deduplication does not work, then you could take a separate approach: define an `internal` interface with all the properties you need in the AzureManaged shared section, and then an internal static class to add these helpers as extension methods. 

But again, if you prefer not to mess around with ways to deduplicate the code that is fine with me also.

Reply:
i will leave it for now thanks Jacob